### ISSUE TITLE
Dragen germline tool error

### ISSUE THREAD

**Issue by skanwal** [umccr/cwl-ica#316]
Our dragen v4 germline tool definition is failing with:

```
ask-container exitCode [1] in Reason [Error].  Message: [Generating run log at /var/log/dragen/dragen_run_1680551660203_7.log\n\nCommand Line: /opt/edico/bin/dragen --partial-reconfig HMM --ignore-version-check true \nDRAGEN Host Software Version 05.121.645.4.0.3 and Bio-IT Processor Version 0x00000000\n\nAutoDetected reference: UNKNOWN\nINFO: Loaded empty agfi\n\n==================================================================\nDownloading HMM HW bitstream (agfi-082bc7358866a1780) - do not interrupt\n==================================================================\n......\nAGFI: Downloaded HW bitstream agfi-082bc7358866a1780\nLICENSE_MSG| Requested streams:empty\nLICENSE_MSG| Empty (No AFUs) request\nPlease set no more than one of fastq_list, fastq_list_rows and bam_input for normal sample\n]\n
```
Ref: trn.b4993d0e133a41b087aed438f636ef73

Introduced in this commit https://github.com/umccr/cwl-ica/commit/5807359ef8a1d6cad655cbcb7488f949ee3102ef#diff-d982735ea5e3abfe93cd80017c44ce9d6fd525db2ae811791de4c3dbc3a54a57R76.

**Comment by alexiswl** [comment]
Ah I'm not sure that this is reproducible, from the log directory

```
gds://development/temp/sehrish/analysis_data/dragen_germline/03042023/work/logs/main/try-1/
```

Looks like two tasks tried to run in the same log directory

```
gds://development/temp/sehrish/analysis_data/dragen_germline/03042023/work/logs/main/try-1/trn.8269ad639dbf481988ee64acc38342b4.json
gds://development/temp/sehrish/analysis_data/dragen_germline/03042023/work/logs/main/try-1/trn.b4993d0e133a41b087aed438f636ef73.json
```

**Comment by alexiswl** [comment]
And the second task `trn.8269ad639dbf481988ee64acc38342b4` completed successfully!

**Comment by alexiswl** [comment]
Please avoid using the same engine parameters for two separate workflow runs

```
ica-get-tasks-from-workflow-history --ica-workflow-run-id wfr.011b74982a1e4ba6af94f1c2502a7aa3
{
  "/main": {
    "task_id": "trn.8269ad639dbf481988ee64acc38342b4",
    "task_name": "main",
    "task_launch_time": "2023-04-03T22:07:47.115492Z",
    "task_completion_time": "2023-04-03T23:45:07.806491Z",
    "task_status": "Succeeded",
    "task_stdout": "gds://development/temp/sehrish/analysis_data/dragen_germline/03042023/work/logs/main/try-1/task-stdout-0.log",
    "task_stderr": "gds://development/temp/sehrish/analysis_data/dragen_germline/03042023/work/logs/main/try-1/task-stderr-0.log"
  }
}
```

```
 ica-get-tasks-from-workflow-history --ica-workflow-run-id wfr.e7a320a59ece4a7ba4a031ca299b2330
{
  "/main": {
    "task_id": "trn.b4993d0e133a41b087aed438f636ef73",
    "task_name": "main",
    "task_launch_time": "2023-04-03T19:37:56.782388Z",
    "task_completion_time": "2023-04-03T19:59:59.643539Z",
    "task_status": "Failed",
    "task_stdout": "gds://development/temp/sehrish/analysis_data/dragen_germline/03042023/work/logs/main/try-1/task-stdout-0.log",
    "task_stderr": "gds://development/temp/sehrish/analysis_data/dragen_germline/03042023/work/logs/main/try-1/task-stderr-0.log"
  }
}
```

**Comment by alexiswl** [comment]
This can't be reproduced since the shell script actually doesn't evaluate that line?

```
gds://development/temp/sehrish/analysis_data/dragen_germline/03042023/work/steps/main/try-1/run-dragen-script.sh
```

```
#!/usr/bin/env bash

# Fail on non-zero exit of subshell
set -euo pipefail

# Run partial reconfig
/opt/edico/bin/dragen \
  --partial-reconfig HMM \
  --ignore-version-check true

# Create directories
mkdir --parents \
  "/ephemeral/ref/" \
  "/ephemeral/intermediate-results/" \
  "2137_germline"

# untar ref data into scratch space
tar \
  --directory "/ephemeral/ref/" \
  --extract \
  --file "/data/cwl-stage-dir/stg9871678f-6b48-4f88-8ad9-989cd64a1c26/hg38-v8-altaware-cnv-anchored.tar.gz"

# Confirm either of fastq_list, fastq_list_rows or bam_input is defined
if [[ "1" -gt "1" ]]; then
  echo "Please set no more than one of fastq_list, fastq_list_rows and bam_input for normal sample" 1>&2
  exit 1
fi

# Run dragen command and import options from cli
eval "/opt/edico/bin/dragen" '"$@"' 
```

But if you run 

```bash
if [[ "1" -gt "1" ]]; then
  echo "foo"
else
  echo "bar"
fi
```

Will yield _bar_ since "1" is not greater than "1"

**Comment by alexiswl** [comment]
Please use cwl-ica create-workflow-submission-template in future and ensure that the __DATE_STR__ is present in the engine parameters, especially for the work directory. This will prevent unreproducible errors like this happening in the future.

**Comment by skanwal** [comment]
Okay - this requires explanation.

First workflow run wfr.e7a320a59ece4a7ba4a031ca299b2330 failed with the error I have pasted in my issue in task trn.b4993d0e133a41b087aed438f636ef73.

The script is your [message](https://github.com/umccr/cwl-ica/issues/316#issuecomment-1496748740) is from second successful task run trn.8269ad639dbf481988ee64acc38342b4 that I executed after making an update https://github.com/umccr/cwl-ica/compare/main...fix/dragen-germline-tool#diff-d982735ea5e3abfe93cd80017c44ce9d6fd525db2ae811791de4c3dbc3a54a57R76 (that I have just pushed).

Re reproducibility of the error, I agree the logs have been overwritten by the latest task run. I am launching a run again so that this could be reproduced.

**Comment by alexiswl** [comment]
> The script is your https://github.com/umccr/cwl-ica/issues/316#issuecomment-1496748740 is from second successful task run trn.8269ad639dbf481988ee64acc38342b4 that I executed after making an update [`main...fix/dragen-germline-tool`#diff-d982735ea5](https://github.com/umccr/cwl-ica/compare/main...fix/dragen-germline-tool#diff-d982735ea5e3abfe93cd80017c44ce9d6fd525db2ae811791de4c3dbc3a54a57R76) (that I have just pushed).

Ah this makes a lot more sense! 

I'd update the logic to be `if [[ ! (js) == "1" ]]; then error; fi`. For the dragen-germline pipeline, we want 1 and ONLY 1 of those inputs. For the somatic pipeline, it can be 0 or 1 since we can run it as tumor only
